Overflow warning for crowdsale calculuations

To quote the build log:
```
In file included from /home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/multiprecision/cpp_int.hpp:1802:0,
                 from mastercore_mdex.h:9,
                 from mastercore.cpp:22:
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/multiprecision/cpp_int/limits.hpp: In instantiation of ‘const int std::numeric_limits<boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> > >::digits10’:
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/detail/lcast_precision.hpp:75:66:   instantiated from ‘const unsigned int boost::detail::lcast_precision<boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> > >::precision_dec’
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/detail/lcast_precision.hpp:79:2:   instantiated from ‘boost::detail::lcast_precision<boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> > >’
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/detail/lcast_precision.hpp:102:32:   instantiated from ‘std::streamsize boost::detail::lcast_get_precision(T*) [with T = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >, std::streamsize = long int]’
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/detail/lcast_precision.hpp:170:5:   instantiated from ‘void boost::detail::lcast_set_precision(std::ios_base&, T*) [with T = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >]’
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/lexical_cast.hpp:2014:17:   instantiated from ‘bool boost::detail::lexical_stream_limited_src<CharT, Traits, RequiresStringbuffer>::shr_using_base_class(InputStreamable&) [with InputStreamable = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >, CharT = char, Traits = std::char_traits<char>, bool RequiresStringbuffer = true]’
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/lexical_cast.hpp:2247:90:   instantiated from ‘bool boost::detail::lexical_stream_limited_src<CharT, Traits, RequiresStringbuffer>::operator>>(InputStreamable&) [with InputStreamable = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >, CharT = char, Traits = std::char_traits<char>, bool RequiresStringbuffer = true]’
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/lexical_cast.hpp:2374:17:   instantiated from ‘static Target boost::detail::lexical_cast_do_cast<Target, Source>::lexical_cast_impl(const Source&) [with Target = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >, Source = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<128u, 128u, (boost::multiprecision::cpp_integer_type)1u, (boost::multiprecision::cpp_int_check_type)0u, void> >]’
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/lexical_cast.hpp:2543:50:   instantiated from ‘Target boost::lexical_cast(const Source&) [with Target = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<> >, Source = boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<128u, 128u, (boost::multiprecision::cpp_integer_type)1u, (boost::multiprecision::cpp_int_check_type)0u, void> >]’
mastercore.cpp:737:96:   instantiated from here
/home/travis/build/dexX7/bitcoin/depends/x86_64-unknown-linux-gnu/include/boost/multiprecision/cpp_int/limits.hpp:133:104: warning: integer overflow in expression [-Woverflow]
```

--- a/src/omnicore/omnicore.cpp
+++ b/src/omnicore/omnicore.cpp
@@ -660,13 +660,13 @@ void calculateFundraiser(uint64_t amtTransfer, unsigned char bonusPerc,
     int128_t bonusPercentage_ = (ebPercentage_ + (precision_ * percentage_precision)) / percentage_precision;
 
     // Calculate the bonus percentage for the issuer
-    int128_t issuerPercentage_ = (int128_t) issuerPerc * precision_ / percentage_precision;
+    int128_t issuerPercentage_ = int128_t(issuerPerc) * precision_ / percentage_precision;
 
     // Precision for bitcoin amounts (satoshi)
     int128_t satoshi_precision_ = 100000000;
 
     // Total tokens including remainders
-    cpp_int createdTokens = boost::lexical_cast<cpp_int>((int128_t) amtTransfer * (int128_t) numProps) * boost::lexical_cast<cpp_int>(bonusPercentage_);
+    cpp_int createdTokens = cpp_int(amtTransfer) * cpp_int(numProps) * cpp_int(bonusPercentage_);
     cpp_int issuerTokens = (createdTokens / (satoshi_precision_ * precision_)) * (issuerPercentage_ / 100) * precision_;
 
     cpp_int createdTokens_int = createdTokens / (precision_ * satoshi_precision_);
@@ -691,7 +691,9 @@ void calculateFundraiser(uint64_t amtTransfer, unsigned char bonusPerc,
     }
 
     // The tokens to credit
-    tokens = std::make_pair(boost::lexical_cast<uint64_t>(createdTokens_int), boost::lexical_cast<uint64_t>(issuerTokens_int));
+    assert(createdTokens_int <= std::numeric_limits<int64_t>::max());
+    assert(issuerTokens_int <= std::numeric_limits<int64_t>::max());
+    tokens = std::make_pair(createdTokens_int.convert_to<uint64_t>(), issuerTokens_int.convert_to<uint64_t>());
 }
 
 // certain transaction types are not live on the network until some specific block height

--- a/src/omnicore/omnicore.cpp
+++ b/src/omnicore/omnicore.cpp
@@ -634,7 +634,7 @@ int64_t before, after;
 
 // calculateFundraiser does token calculations per transaction
 // calcluateFractional does calculations for missed tokens
-void calculateFundraiser(unsigned short int propType, uint64_t amtTransfer, unsigned char bonusPerc,
+void calculateFundraiser(uint64_t amtTransfer, unsigned char bonusPerc,
         uint64_t fundraiserSecs, uint64_t currentSecs, uint64_t numProps, unsigned char issuerPerc, uint64_t totalTokens,
         std::pair<uint64_t, uint64_t>& tokens, bool &close_crowdsale)
 {
@@ -3939,16 +3939,9 @@ int CMPTransaction::logicMath_SimpleSend()
                 }
 
                 // Calculate the amounts to credit for this fundraiser
-                calculateFundraiser(sp.prop_type,
-                        nValue,
-                        sp.early_bird,
-                        sp.deadline,
-                        (uint64_t) blockTime,
-                        sp.num_tokens,
-                        sp.percentage,
-                        getTotalTokens(pcrowdsale->getPropertyId()),
-                        tokens,
-                        close_crowdsale);
+                calculateFundraiser(nValue, sp.early_bird, sp.deadline, (uint64_t) blockTime,
+                        sp.num_tokens, sp.percentage, getTotalTokens(pcrowdsale->getPropertyId()),
+                        tokens, close_crowdsale);
 
                 // Update the crowdsale object
                 pcrowdsale->incTokensUserCreated(tokens.first);

--- a/src/omnicore/omnicore.cpp
+++ b/src/omnicore/omnicore.cpp
@@ -3889,109 +3889,89 @@ std::string new_global_alert_message;
 
 int CMPTransaction::logicMath_SimpleSend()
 {
-int rc = PKT_ERROR_SEND -1000;
-int invalid = 0;  // unused
+    int rc = PKT_ERROR_SEND -1000;
 
-      if (!isTransactionTypeAllowed(block, property, type, version)) return (PKT_ERROR_SEND -22);
+    if (!isTransactionTypeAllowed(block, property, type, version)) {
+        return PKT_ERROR_SEND -22;
+    }
 
-      if (sender.empty()) ++invalid;
-      // special case: if can't find the receiver -- assume sending to itself !
-      // may also be true for BTC payments........
-      // TODO: think about this..........
-      if (receiver.empty())
-      {
+    // Special case: if can't find the receiver -- assume send to self!
+    if (receiver.empty()) {
         receiver = sender;
-      }
-      if (receiver.empty()) ++invalid;
+    }
 
-      // insufficient funds check & return
-      if (!update_tally_map(sender, property, - nValue, BALANCE))
-      {
-        return (PKT_ERROR -111);
-      }
+    // Cancel send, if the sender has sufficient funds
+    if (!update_tally_map(sender, property, -nValue, BALANCE)) {
+        return PKT_ERROR -111;
+    }
 
-      update_tally_map(receiver, property, nValue, BALANCE);
+    update_tally_map(receiver, property, nValue, BALANCE);
 
-      // is there a crowdsale running from this recepient ?
-      {
-      CMPCrowd *crowd;
+    // Is there an active crowdsale running from this recepient?
+    {
+        CMPCrowd* pcrowdsale = getCrowd(receiver);
 
-        crowd = getCrowd(receiver);
+        if (pcrowdsale && pcrowdsale->getCurrDes() == property) {
+            CMPSPInfo::Entry sp;
+            bool spFound = _my_sps->getSP(pcrowdsale->getPropertyId(), sp);
 
-        if (crowd && (crowd->getCurrDes() == property) )
-        {
-          CMPSPInfo::Entry sp;
-          bool spFound = _my_sps->getSP(crowd->getPropertyId(), sp);
+            PrintToLog("INVESTMENT SEND to Crowdsale Issuer: %s\n", receiver);
 
-          PrintToLog("INVESTMENT SEND to Crowdsale Issuer: %s\n", receiver);
-          
-          if (spFound)
-          {
-            //init this struct
-            std::pair <uint64_t,uint64_t> tokens;
-            //pass this in by reference to determine if max_tokens has been reached
-            bool close_crowdsale = false; 
-            //get txid
-            string sp_txid =  sp.txid.GetHex();
+            if (spFound) {
+                // Holds the tokens to be credited to the sender and receiver
+                std::pair<uint64_t, uint64_t> tokens;
 
-            //Units going into the calculateFundraiser function must
-            //match the unit of the fundraiser's property_type.
-            //By default this means Satoshis in and satoshis out.
-            //In the condition that your fundraiser is Divisible,
-            //but you are accepting indivisible tokens, you must
-            //account for 1.0 Div != 1 Indiv but actually 1.0 Div == 100000000 Indiv.
-            //The unit must be shifted or your values will be incorrect,
-            //that is what we check for below.
-            if ( !(isPropertyDivisible(property)) ) {
-              nValue = nValue * 1e8;
+                // Passed by reference to determine, if max_tokens has been reached
+                bool close_crowdsale = false;
+
+                // Units going into the calculateFundraiser function must
+                // match the unit of the fundraiser's property_type.
+                // By default this means satoshis in and satoshis out.
+                // In the condition that the fundraiser is divisible,
+                // but indivisible tokens are accepted, it must account for
+                // 1.0 Div != 1 Indiv, but actually 1.0 Div == 100000000 Indiv.
+                // The unit must be shifted or the values will be incorrect,
+                // which is what is checked below.
+                if (!isPropertyDivisible(property)) {
+                    nValue = nValue * 1e8;
+                }
+
+                // Calculate the amounts to credit for this fundraiser
+                calculateFundraiser(sp.prop_type,
+                        nValue,
+                        sp.early_bird,
+                        sp.deadline,
+                        (uint64_t) blockTime,
+                        sp.num_tokens,
+                        sp.percentage,
+                        getTotalTokens(pcrowdsale->getPropertyId()),
+                        tokens,
+                        close_crowdsale);
+
+                // Update the crowdsale object
+                pcrowdsale->incTokensUserCreated(tokens.first);
+                pcrowdsale->incTokensIssuerCreated(tokens.second);
+
+                // Data to pass to txFundraiserData
+                uint64_t txdata[] = {(uint64_t) nValue, (uint64_t) blockTime, (uint64_t) tokens.first, (uint64_t) tokens.second};
+                std::vector<uint64_t> txDataVec(txdata, txdata + sizeof (txdata) / sizeof (txdata[0]));
+
+                // Insert data about crowdsale participation
+                pcrowdsale->insertDatabase(txid.GetHex(), txDataVec);
+
+                // Credit tokens for this fundraiser
+                update_tally_map(sender, pcrowdsale->getPropertyId(), tokens.first, BALANCE);
+                update_tally_map(receiver, pcrowdsale->getPropertyId(), tokens.second, BALANCE);
+
+                // Close crowdsale, if we hit MAX_TOKENS
+                if (close_crowdsale) {
+                    eraseMaxedCrowdsale(receiver, blockTime, block);
+                }
             }
-
-            //file_log("\nValues going into calculateFundraiser(): hexid %s nValue %lu earlyBird %d deadline %lu blockTime %ld numProps %lu issuerPerc %d \n", txid.GetHex().c_str(), nValue, sp.early_bird, sp.deadline, (uint64_t) blockTime, sp.num_tokens, sp.percentage);
-
-            // calc tokens per this fundraise
-            calculateFundraiser(sp.prop_type,         //u short
-                                nValue,               // u int 64
-                                sp.early_bird,        // u char
-                                sp.deadline,          // u int 64
-                                (uint64_t) blockTime, // int 64
-                                sp.num_tokens,      // u int 64
-                                sp.percentage,        // u char
-                                getTotalTokens(crowd->getPropertyId()),
-                                tokens,
-                                close_crowdsale);
-
-            //file_log(mp_fp,"\n before incrementing global tokens user: %ld issuer: %ld\n", crowd->getUserCreated(), crowd->getIssuerCreated());
-            
-            //getIssuerCreated() is passed into calcluateFractional() at close
-            //getUserCreated() is a convenient way to get user created during a crowdsale
-            crowd->incTokensUserCreated(tokens.first);
-            crowd->incTokensIssuerCreated(tokens.second);
-            
-            //file_log(mp_fp,"\n after incrementing global tokens user: %ld issuer: %ld\n", crowd->getUserCreated(), crowd->getIssuerCreated());
-            
-            //init data to pass to txFundraiserData
-            uint64_t txdata[] = { (uint64_t) nValue, (uint64_t) blockTime, (uint64_t) tokens.first, (uint64_t) tokens.second };
-            
-            std::vector<uint64_t> txDataVec(txdata, txdata + sizeof(txdata)/sizeof(txdata[0]) );
-
-            //insert data
-            crowd->insertDatabase(txid.GetHex().c_str(), txDataVec  );
-
-            //file_log(mp_fp,"\nValues coming out of calculateFundraiser(): hex %s: Tokens created, Tokens for issuer: %ld %ld\n",txid.GetHex().c_str(), tokens.first, tokens.second);
-
-            //update sender/rec
-            update_tally_map(sender, crowd->getPropertyId(), tokens.first, BALANCE);
-            update_tally_map(receiver, crowd->getPropertyId(), tokens.second, BALANCE);
-
-            // close crowdsale if we hit MAX_TOKENS
-            if( close_crowdsale ) {
-              eraseMaxedCrowdsale(receiver, blockTime, block);
-            }
-          }
         }
-      }
+    }
 
-      rc = 0;
+    rc = 0;
 
     return rc;
 }

--- a/src/omnicore/omnicore.cpp
+++ b/src/omnicore/omnicore.cpp
@@ -634,62 +634,62 @@ int64_t before, after;
 
 //calculateFundraiser does token calculations per transaction
 //calcluateFractional does calculations for missed tokens
-void calculateFundraiser(unsigned short int propType, uint64_t amtTransfer, unsigned char bonusPerc, 
-  uint64_t fundraiserSecs, uint64_t currentSecs, uint64_t numProps, unsigned char issuerPerc, uint64_t totalTokens, 
-  std::pair<uint64_t, uint64_t>& tokens, bool &close_crowdsale )
+void calculateFundraiser(unsigned short int propType, uint64_t amtTransfer, unsigned char bonusPerc,
+        uint64_t fundraiserSecs, uint64_t currentSecs, uint64_t numProps, unsigned char issuerPerc, uint64_t totalTokens,
+        std::pair<uint64_t, uint64_t>& tokens, bool &close_crowdsale)
 {
-  int128_t weeks_sec_ = 604800L;
-  //define weeks in seconds
-  int128_t precision_ = 1000000000000L;
-  //define precision for all non-bitcoin values (bonus percentages, for example)
-  int128_t percentage_precision = 100L;
-  //define precision for all percentages (10/100 = 10%)
+    int128_t weeks_sec_ = 604800L;
+    //define weeks in seconds
+    int128_t precision_ = 1000000000000L;
+    //define precision for all non-bitcoin values (bonus percentages, for example)
+    int128_t percentage_precision = 100L;
+    //define precision for all percentages (10/100 = 10%)
 
-  //calcluate the bonusseconds
-  int128_t bonusSeconds_ = fundraiserSecs - currentSecs;
+    //calcluate the bonusseconds
+    int128_t bonusSeconds_ = fundraiserSecs - currentSecs;
 
-  int128_t weeks_ = (bonusSeconds_ / weeks_sec_) * precision_ + ( (bonusSeconds_ % weeks_sec_ ) * precision_) / weeks_sec_;
-  //calculate the whole number of weeks to apply bonus
+    int128_t weeks_ = (bonusSeconds_ / weeks_sec_) * precision_ + ((bonusSeconds_ % weeks_sec_) * precision_) / weeks_sec_;
+    //calculate the whole number of weeks to apply bonus
 
-  int128_t ebPercentage_ = weeks_ * bonusPerc;
-  //calculate the earlybird percentage to be applied
+    int128_t ebPercentage_ = weeks_ * bonusPerc;
+    //calculate the earlybird percentage to be applied
 
-  int128_t bonusPercentage_ = (ebPercentage_ + (precision_ * percentage_precision) ) / percentage_precision; 
-  //calcluate the bonus percentage to apply up to 'percentage_precision' number of digits
+    int128_t bonusPercentage_ = (ebPercentage_ + (precision_ * percentage_precision)) / percentage_precision;
+    //calcluate the bonus percentage to apply up to 'percentage_precision' number of digits
 
-  int128_t issuerPercentage_ = (int128_t)issuerPerc * precision_ / percentage_precision;
+    int128_t issuerPercentage_ = (int128_t) issuerPerc * precision_ / percentage_precision;
 
-  int128_t satoshi_precision_ = 100000000;
-  //define the precision for bitcoin amounts (satoshi)
-  //declare used variables for total created tokens
+    int128_t satoshi_precision_ = 100000000;
+    //define the precision for bitcoin amounts (satoshi)
+    //declare used variables for total created tokens
 
-  //declare used variables for total issuer tokens
+    //declare used variables for total issuer tokens
 
-  cpp_int createdTokens = boost::lexical_cast<cpp_int>((int128_t)amtTransfer*(int128_t)numProps)* boost::lexical_cast<cpp_int>(bonusPercentage_);
+    cpp_int createdTokens = boost::lexical_cast<cpp_int>((int128_t) amtTransfer * (int128_t) numProps) * boost::lexical_cast<cpp_int>(bonusPercentage_);
 
-  cpp_int issuerTokens = (createdTokens / (satoshi_precision_ * precision_ )) * (issuerPercentage_ / 100) * precision_;
+    cpp_int issuerTokens = (createdTokens / (satoshi_precision_ * precision_)) * (issuerPercentage_ / 100) * precision_;
 
-  //total tokens including remainders
+    //total tokens including remainders
 
-  cpp_int createdTokens_int = createdTokens / (precision_ * satoshi_precision_);
-  cpp_int issuerTokens_int = issuerTokens / (precision_ * satoshi_precision_ * 100 );
-  cpp_int newTotalCreated = totalTokens + createdTokens_int  + issuerTokens_int;
+    cpp_int createdTokens_int = createdTokens / (precision_ * satoshi_precision_);
+    cpp_int issuerTokens_int = issuerTokens / (precision_ * satoshi_precision_ * 100);
+    cpp_int newTotalCreated = totalTokens + createdTokens_int + issuerTokens_int;
 
-  if ( newTotalCreated > MAX_INT_8_BYTES) {
-    cpp_int maxCreatable = MAX_INT_8_BYTES - totalTokens;
+    if (newTotalCreated > MAX_INT_8_BYTES) {
+        cpp_int maxCreatable = MAX_INT_8_BYTES - totalTokens;
 
-    cpp_int created = createdTokens_int + issuerTokens_int;
-    cpp_int ratio = (created * precision_ * satoshi_precision_) / maxCreatable;
-  
-    issuerTokens_int = (issuerTokens_int * precision_ * satoshi_precision_)/ratio;
-    //calcluate the ratio of tokens for what we can create and apply it
-    createdTokens_int = MAX_INT_8_BYTES - issuerTokens_int ;
-    //give the rest to the user
+        cpp_int created = createdTokens_int + issuerTokens_int;
+        cpp_int ratio = (created * precision_ * satoshi_precision_) / maxCreatable;
 
-    close_crowdsale = true; //close up the crowdsale after assigning all tokens
-  }
-  tokens = std::make_pair(boost::lexical_cast<uint64_t>(createdTokens_int) , boost::lexical_cast<uint64_t>(issuerTokens_int));
-  //give tokens
+        issuerTokens_int = (issuerTokens_int * precision_ * satoshi_precision_) / ratio;
+        //calcluate the ratio of tokens for what we can create and apply it
+        createdTokens_int = MAX_INT_8_BYTES - issuerTokens_int;
+        //give the rest to the user
+
+        close_crowdsale = true; //close up the crowdsale after assigning all tokens
+    }
+    tokens = std::make_pair(boost::lexical_cast<uint64_t>(createdTokens_int), boost::lexical_cast<uint64_t>(issuerTokens_int));
+    //give tokens
 }
 
 // certain transaction types are not live on the network until some specific block height

--- a/src/omnicore/omnicore.cpp
+++ b/src/omnicore/omnicore.cpp
@@ -632,44 +632,42 @@ int64_t before, after;
   return bRet;
 }
 
-//calculateFundraiser does token calculations per transaction
-//calcluateFractional does calculations for missed tokens
+// calculateFundraiser does token calculations per transaction
+// calcluateFractional does calculations for missed tokens
 void calculateFundraiser(unsigned short int propType, uint64_t amtTransfer, unsigned char bonusPerc,
         uint64_t fundraiserSecs, uint64_t currentSecs, uint64_t numProps, unsigned char issuerPerc, uint64_t totalTokens,
         std::pair<uint64_t, uint64_t>& tokens, bool &close_crowdsale)
 {
+    // Weeks in seconds
     int128_t weeks_sec_ = 604800L;
-    //define weeks in seconds
-    int128_t precision_ = 1000000000000L;
-    //define precision for all non-bitcoin values (bonus percentages, for example)
-    int128_t percentage_precision = 100L;
-    //define precision for all percentages (10/100 = 10%)
 
-    //calcluate the bonusseconds
+    // Precision for all non-bitcoin values (bonus percentages, for example)
+    int128_t precision_ = 1000000000000L;
+
+    // Precision for all percentages (10/100 = 10%)
+    int128_t percentage_precision = 100L;
+
+    // Calculate the bonus seconds
     int128_t bonusSeconds_ = fundraiserSecs - currentSecs;
 
+    // Calculate the whole number of weeks to apply bonus
     int128_t weeks_ = (bonusSeconds_ / weeks_sec_) * precision_ + ((bonusSeconds_ % weeks_sec_) * precision_) / weeks_sec_;
-    //calculate the whole number of weeks to apply bonus
 
+    // Calculate the earlybird percentage to be applied
     int128_t ebPercentage_ = weeks_ * bonusPerc;
-    //calculate the earlybird percentage to be applied
 
+    // Calcluate the bonus percentage to apply up to percentage_precision number of digits
     int128_t bonusPercentage_ = (ebPercentage_ + (precision_ * percentage_precision)) / percentage_precision;
-    //calcluate the bonus percentage to apply up to 'percentage_precision' number of digits
 
+    // Calculate the bonus percentage for the issuer
     int128_t issuerPercentage_ = (int128_t) issuerPerc * precision_ / percentage_precision;
 
+    // Precision for bitcoin amounts (satoshi)
     int128_t satoshi_precision_ = 100000000;
-    //define the precision for bitcoin amounts (satoshi)
-    //declare used variables for total created tokens
 
-    //declare used variables for total issuer tokens
-
+    // Total tokens including remainders
     cpp_int createdTokens = boost::lexical_cast<cpp_int>((int128_t) amtTransfer * (int128_t) numProps) * boost::lexical_cast<cpp_int>(bonusPercentage_);
-
     cpp_int issuerTokens = (createdTokens / (satoshi_precision_ * precision_)) * (issuerPercentage_ / 100) * precision_;
-
-    //total tokens including remainders
 
     cpp_int createdTokens_int = createdTokens / (precision_ * satoshi_precision_);
     cpp_int issuerTokens_int = issuerTokens / (precision_ * satoshi_precision_ * 100);
@@ -677,19 +675,23 @@ void calculateFundraiser(unsigned short int propType, uint64_t amtTransfer, unsi
 
     if (newTotalCreated > MAX_INT_8_BYTES) {
         cpp_int maxCreatable = MAX_INT_8_BYTES - totalTokens;
-
         cpp_int created = createdTokens_int + issuerTokens_int;
+
+        // Calcluate the ratio of tokens for what we can create and apply it
         cpp_int ratio = (created * precision_ * satoshi_precision_) / maxCreatable;
 
+        // The tokens for the issuer
         issuerTokens_int = (issuerTokens_int * precision_ * satoshi_precision_) / ratio;
-        //calcluate the ratio of tokens for what we can create and apply it
-        createdTokens_int = MAX_INT_8_BYTES - issuerTokens_int;
-        //give the rest to the user
 
-        close_crowdsale = true; //close up the crowdsale after assigning all tokens
+        // The tokens for the user
+        createdTokens_int = MAX_INT_8_BYTES - issuerTokens_int;
+
+        // Close the crowdsale after assigning all tokens
+        close_crowdsale = true;
     }
+
+    // The tokens to credit
     tokens = std::make_pair(boost::lexical_cast<uint64_t>(createdTokens_int), boost::lexical_cast<uint64_t>(issuerTokens_int));
-    //give tokens
 }
 
 // certain transaction types are not live on the network until some specific block height

